HelpSource/Classes/SimpleNumber.schelp

```diff
@@ -360,7 +360,184 @@ method:: booleanValue
+::
+
+teletype::
+-> true
+::
+
+code::
+0.0 === -0.0 // checking for identity
+::
+
+teletype::
+-> false
+::
+
+Discussion::
+
+The same number will always result in the same emphasis::hash index:: and in the same emphasis::identity hash index:::
+
+code::
+// Evaluate the following codes more than once and compare the results.
+0.hash
+0.identityHash
+0.0.hash
+0.0.identityHash
+-0.0.hash
+-0.0.identityHash
+
+// Or check that each of the following pieces of code always returns true:
+0.hash == 0.hash
+0.identityHash == 0.identityHash
+0.0.hash == 0.0.hash
+0.0.identityHash == 0.0.identityHash
+-0.0.hash == -0.0.hash
+-0.0.identityHash == -0.0.identityHash
+::
+
+list::
+
+## The equality and identity of an integer and a float with the same value as the integer:
+
+If an integer and a float are equal,
+then their emphasis::hash table indices:: are equal and identical:
+code::
+0.hash == 0.0.hash
+::
+teletype::
+-> true
+::
+code::
+0.hash === 0.0.hash
+::
+teletype::
+-> true
+::
+However, although an integer and a float are equal,
+their emphasis::identity hash table indices:: are
+neither equal nor identical:
+
+code::
+0.identityHash == 0.0.identityHash
+::
+teletype::
+-> false
+::
+code::
+0.identityHash === 0.0.identityHash
+::
+teletype::
+-> false
+::
+## The inaccuracy of equality between human readable decimals:
+
+0.1 plus 0.1 plus 0.1 equals 0.3:
+code::
+0.1 + 0.1 + 0.1
+::
+teletype::
+-> 0.3
+::
+However, sclang returns false when checking for equality of the two:
+
+code::
+0.3 == (0.1 + 0.1 + 0.1)
+::
+teletype::
+-> false
+::
+This is due to the imprecise decimal representation of float
+because the 64-bit SuperCollider uses emphasis::64-bit double precision floats:: on the client (sclang) and emphasis::32-bit floats:: on the server (scsynth and supernova).
+The human-readable decimal representation of the float value may not be accurate.
+To check the more precise hexadecimal representation of float in sclang, the code::.dump:: method (see: link::Classes/Object#-dump::) can be used:
+
+code::
+0.3.dump
+::
+teletype::
+   Float 0.300000   33333333 3FD33333
+-> 0.3
+::
+code::
+(0.1 + 0.1 + 0.1).dump
+::
+teletype::
+   Float 0.300000   33333334 3FD33333
+-> 0.3
+::
+## The equality and identity of 0.0 and -0.0:
+
+The existence of 0.0 and -0.0 in computing is a result of the way floating point numbers are represented. They are equal in value:
+code::
+0.0 == -0.0
+::
+teletype::
+-> true
+::
+However, their identities are not the same:
+code::
+0.0 === -0.0
+::
+teletype::
+-> false
+::
+code::
+0.0.dump
+::
+teletype::
+   Float 0.000000   00000000 00000000
+-> 0.0
+::
+code::
+-0.0.dump
+::
+teletype::
+   Float -0.000000   00000000 80000000
+-> -0.0
+::
+Consequently, their emphasis::hash table indices:: and emphasis::identity hash indices:: are not the same:
+code::
+0.0.hash == -0.0.hash
+::
+teletype::
+-> false
+::
+code::
+0.0.identityHash == -0.0.identityHash
+::
+teletype::
+-> false
+::
+::
+
+
```

Comment:
I think this should all go and just be replaced by 'checks for equality.' Then link to the equality file you wrote eariler.

Reply:
Thanks!
---
Just a note:

It was originally in the Object.schelp file:
https://github.com/supercollider/supercollider/files/13809448/Object._.SuperCollider.3.14.0-dev.Help.pdf (from https://github.com/supercollider/supercollider/pull/6174's initial PR)

I moved it here according to the following review:
https://github.com/supercollider/supercollider/pull/6174#discussion_r1439773347
---
However, I think you want to remove this part completely. Am I correct?

Reply:
Ah I didn't understand at first, but this is about how Float and Integer can't be interchanged as keys in Event. You should be able to explain this in less than 100 lines.

I'd say something like.

---

Although link::Classes/Float:: and link::Classes/Integer:: can be equal, `1.0 == 1 // true`, they are **never** identical. When using numbers as keys in an link::Classes/IdentityDictionary:: or link::Classes/Event:: extra case should be taken to convert them to the same type.
```
e = (1 : \a);
e[1] // -> \a
e[1.0] // -> nil
e[1.0.asInteger] // -> \a
``` 
Further, due to floating point inaccuracies it is advisable to **never** use link::Classes/Float:: as keys.
```
e = (0.3 : \a);
e[0.3] // -> \a
e[0.1 + 0.1 + 0.1] // nil ... ??
```
For more information see (insert link to conversation about floating point inaccuracies).

---

There could also be a link to a different section exampling float point inaccuracies, there is no need to explain what this is here, just stick to the bare minimum.

Reply:
OK, I will do it in the new file. Is it ok?